### PR TITLE
CLI can now scan and zip all content from a particular path

### PR DESCRIPTION
**Pull request by rafaveira3**
## Description

This PR is part of a series of changes that our DEV team is planning to do to create a new huskyCI CLI. A new architecture is being discussed so that we transfer the responsibility of cloning and sending the code to be analyzed to the huskyCI API to the user(terminal) or CI(stage jobs).  

The new CLI approach involves 5 stages and this PR includes the **2 initial ones**!
* Scan: Analyze the path in which the code will be sent to the huskyCI API so that users have an idea of how many files are being tested and also which languages will be supported during the analysis.
* Compress: All files will be compressed into a `GUID.zip` file to be sent to the API.
* Send: The CLI will send the generated file and also remove it after the huskyCI successfully receives it.
* Check: It will be a worker to track the status of the analysis. 
* Print: The final step includes analyzing each result and printing the results to STDOUT and returning a status coded based on them.

## Changes

* Vendor folder and go mod dependencies were updated.
* The old logic to start a new analysis has been removed from the `cli`.
* Analysis package has been introduced with all the 5 methods mentioned above, with only the 2 initial ones working.

## Testing

```sh
✔ [16:05] rafaelsantos@labs:~/go/src/github.com/globocom/huskyCI (refact-cli)
$ make build-cli
✔ [16:05] rafaelsantos@labs:~/go/src/github.com/globocom/huskyCI (refact-cli)
$ ./huskyci-client run .

[HUSKYCI] Scanning your code from /Users/rafaelsantos/go/src/github.com/globocom/huskyCI
[HUSKYCI] 2425 files found
[HUSKYCI] JavaScript -> [huskyci/npmaudit huskyci/yarnaudit]
[HUSKYCI] Generic -> [huskyci/gitleaks]
[HUSKYCI] Go -> [huskyci/gosec]
[HUSKYCI] Scanned!

[HUSKYCI] Compressing your code...
[HUSKYCI] Compressed!  /Users/rafaelsantos/e1f7937b-7788-4179-b38b-2199bfc47ae9.zip 49.5 MB

[HUSKYCI] Sending your code to the huskyCI API at...
[HUSKYCI] Sent!

[HUSKYCI] Checking if the analysis has already finished...
[HUSKYCI] Checked!

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ build-client-linux:
 
 ## Builds CLI to the executable file huskyci-client
 build-cli:
-	cd cli && $(GO) build -o "$(HUSKYCICLIENTBIN)" main.go
+	cd cli && $(GO) build -o "$(HUSKYCICLIENTBIN)" main.go && mv "$(HUSKYCICLIENTBIN)" ../
```
Comment (joserenatosilva) [inline Makefile]: Suggestion:
`$(GO) build -o "$(HUSKYCICLIENTBIN)" cli/main.go`

**Comment by joserenatosilva** [comment]
```sh
$ ./huskyci-client run .
Client created config folder: /Users/joserenato.junior/.huskyci
Client error creating config file: /Users/joserenato.junior/.huskyci/config.yaml (open /Users/joserenato.junior/.huskyci/config.yaml: not a directory)
```

I don't know why but got this error executing for the first time. I deleted everything and tried again without any issue. I'm currently investigating this.

File: Makefile (hunk 1 above)
Comment (rafaveira3) [reply Makefile]: Great! Thanks for this tip. Did it here 9d9b5e7.

File: cli/main.go [hunk 2]
```diff
@@ -5,9 +5,15 @@
...
 import (
+	"fmt"
+	"os"
+
 	"github.com/globocom/huskyCI/cli/cmd"
 )
 
 func main() {
-	cmd.Execute()
+	if err := cmd.Execute(); err != nil {
```
[the remark this answers is not in the document]
Comment (rafaveira3) [reply cli/main.go]: Good catch, @joserenatosilva! I have changed it here b8a2a26126d343ccb7f88d5b6dadfd65ca237d06. Thanks! 🙃

**Pull request merged**